Assets/Mirror/Weaver/Processors/NetworkBehaviourProcessor.cs

```diff
@@ -347,22 +346,38 @@ void GenerateSerialization()
             serWorker.Append(serWorker.Create(OpCodes.Ldarg_2)); // forceAll
             serWorker.Append(serWorker.Create(OpCodes.Brfalse, initialStateLabel));
 
+            int netIdFieldCounter = 0;
             foreach (FieldDefinition syncVar in m_SyncVars)
             {
-                // Generates a writer call for each sync variable
-                serWorker.Append(serWorker.Create(OpCodes.Ldarg_1)); // writer
-                serWorker.Append(serWorker.Create(OpCodes.Ldarg_0)); // this
-                serWorker.Append(serWorker.Create(OpCodes.Ldfld, syncVar));
-                MethodReference writeFunc = Weaver.GetWriteFunc(syncVar.FieldType);
-                if (writeFunc != null)
+                if (syncVar.FieldType.FullName == Weaver.gameObjectType.FullName ||
+                    syncVar.FieldType.FullName == Weaver.NetworkIdentityType.FullName)
                 {
-                    serWorker.Append(serWorker.Create(OpCodes.Call, writeFunc));
+                    // GameObject/NetworkIdentity SyncVar - write generated netId var
+                    FieldDefinition netIdField = m_SyncVarNetIds[netIdFieldCounter];
+                    netIdFieldCounter += 1;
+
+                    serWorker.Append(serWorker.Create(OpCodes.Ldarg_1)); // writer
+                    serWorker.Append(serWorker.Create(OpCodes.Ldarg_0)); // this
+                    serWorker.Append(serWorker.Create(OpCodes.Ldfld, netIdField));
+                    serWorker.Append(serWorker.Create(OpCodes.Callvirt, Weaver.NetworkWriterWritePacked32));
```

Comment:
still broken for scene objects.    In this case,  the netidField will be 0,  even though the field points to a valid object with a netid